addons/stock/models/stock_move.py

```diff
@@ -1701,9 +1709,11 @@ def _update_reserved_quantity(self, need, location_id, lot_id=None, package_id=N
                 if self.product_id.tracking == 'serial' and (self.picking_type_id.use_create_lots or self.picking_type_id.use_existing_lots):
                     vals_list = self._add_serial_move_line_to_vals_list(reserved_quant, quantity)
                     if vals_list:
-                        self.env['stock.move.line'].with_context(reserved_quant=reserved_quant).create(vals_list)
+                        move_line_vals += vals_list
                 else:
-                    self.env['stock.move.line'].with_context(reserved_quant=reserved_quant).create(self._prepare_move_line_vals(quantity=quantity, reserved_quant=reserved_quant))
+                    move_line_vals.append(self._prepare_move_line_vals(quantity=quantity, reserved_quant=reserved_quant))
```

Comment:
you can also remove the `context.get('reserved_quant') in stock_move_line.py to avoid confusion

Reply:
reserved_quant still used but not in this case since we are batch creating the stock.move.line